apollo-ios/Sources/Apollo/GraphQLResult.swift

```diff
@@ -1,38 +1,90 @@
+import Foundation
 #if !COCOAPODS
 import ApolloAPI
 #endif
 
 /// Represents the result of a GraphQL operation.
 public struct GraphQLResult<Data: RootSelectionSet> {
 
+  /// Represents source of data
+  public enum Source: Hashable {
+    case cache
+    case server
+  }
+
   /// The typed result data, or `nil` if an error was encountered that prevented a valid response.
   public let data: Data?
   /// A list of errors, or `nil` if the operation completed without encountering any errors.
   public let errors: [GraphQLError]?
   /// A dictionary which services can use however they see fit to provide additional information to clients.
   public let extensions: [String: AnyHashable]?
-
-  /// Represents source of data
-  public enum Source: Hashable {
-    case cache
-    case server
-  }
   /// Source of data
   public let source: Source
 
   let dependentKeys: Set<CacheKey>?
 
-  public init(data: Data?,
-              extensions: [String: AnyHashable]?,
-              errors: [GraphQLError]?,
-              source: Source,
-              dependentKeys: Set<CacheKey>?) {
+  public init(
+    data: Data?,
+    extensions: [String: AnyHashable]?,
+    errors: [GraphQLError]?,
+    source: Source,
+    dependentKeys: Set<CacheKey>?
+  ) {
     self.data = data
     self.extensions = extensions
     self.errors = errors
     self.source = source
     self.dependentKeys = dependentKeys
   }
+
+  func merging(_ incrementalResult: IncrementalGraphQLResult) throws -> GraphQLResult<Data> {
+    var mergedData = self.data
```

Comment:
We can consider using a protocol or function that you pass a closure to in order to clean up the repetitive part of this merging.

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/34c171ed6be6c04bfe0b7081b4548f6d9b8c8566.